### DISCUSSION TITLE
Generate forms

### DISCUSSION THREAD

**Discussion by iturtev** [firecmsco/firecms#472] [Q&A]
Hello,
Is there a way to generate a form using form components? 
The use case is: I have a text with placeholders. For each placeholder I want to generate a field based on the type (date, reference, text, etc.). After the user fills in the form, the values will be replaced in the text.

**Comment by fgatti675** [comment]
You mean replace the side form with a custom one right?

**Comment by iturtev** [comment]
No, I don't want to replace the side form, but to generate a form based on the user input in a custom view. I can generate the form itself, what I need is how to use the cms field types. For example, I want to create a custom form (not for CRUD, but a custom functionality) that will have date fields, reference fields, selects/multiselects, boolean, etc.
